### ISSUE TITLE
Add integration fails with “Failed to setup”

### ISSUE THREAD

**Issue by tobbensoft** [frenck/spook#570]
### What version of Spook are you using?

2.1.1

### What version of Home Assistant are you using?

2024.1.3

### The problem

Trying to setup the integration, but it fails with “Failed to setup”

![image](https://github.com/frenck/spook/assets/46381052/b61c2c8a-8699-4df6-a5e9-c62308c451b4)

Tried to remove and reinstall with same results.

Attaching logs.

Any tips on what to check? Or is it a bug?

### Anything in the logs? Paste it here!

```plain text
Logger: homeassistant.config_entries
Source: config_entries.py:406
First occurred: 15:54:25 (1 occurrences)
Last logged: 15:54:25

Error setting up entry Not your homie for spook
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 406, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/spook/__init__.py", line 90, in async_setup_entry
    await services.async_setup()
  File "/config/custom_components/spook/services.py", line 288, in async_setup
    self.async_register_service(service)
  File "/config/custom_components/spook/services.py", line 293, in async_register_service
    service.async_register()
  File "/config/custom_components/spook/services.py", line 168, in async_register
    for platform in self.hass.data[DATA_ENTITY_PLATFORM][self.domain]
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
KeyError: 'timer'
```

**Comment by schford** [comment]
Same issue for me.

**Comment by manonstreet** [comment]
Same here.  I added `timer:` to my configuration.yaml, restarted HA and the issue resolved.

**Comment by tobbensoft** [comment]
@manonstreet, that worked here as well.  Thanks!

**Comment by lacrima08** [comment]
@manonstreet Thank you ! It's working !

**Comment by frenck** [comment]
This has been fixed with the release of Home Assistant 2024.2, which now always loads up the timer.

../Frenck

**Comment by heibertelf** [comment]
> This has been fixed with the release of Home Assistant 2024.2, which now always loads up the timer.
> 
> ../Frenck

Hi! I'm not very advanced, so I'm not sure what to do next yet. Here's what I have:
![image](https://github.com/frenck/spook/assets/80182970/675fb558-c2bc-4caf-b1d8-273da0e71b16)
![image](https://github.com/frenck/spook/assets/80182970/b1bd3ec1-ccc4-4a09-9d7a-6957cd8786b2)

```
Logger: homeassistant.config_entries
Source: config_entries.py:444
First occurred: 5:42:37 PM (1 occurrences)
Last logged: 5:42:37 PM

Error setting up entry Not your homie for spook
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 444, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/spook/__init__.py", line 38, in async_setup_entry
    if await hass.async_add_executor_job(link_sub_integrations, hass):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/spook/util.py", line 75, in link_sub_integrations
    dest.symlink_to(src)
  File "/usr/local/lib/python3.12/pathlib.py", line 1386, in symlink_to
    os.symlink(target, self, target_is_directory)
FileExistsError: [Errno 17] File exists: '/config/custom_components/spook/integrations/spook_inverse' -> '/config/custom_components/spook_inverse'

```

**Comment by diplix** [comment]
> config/custom_components/spook/integrations/spook_inverse

i had this problem after moving my installation to docker. there was an old symlink in `/config/custom_components`  with a wrong path which i needed to delete. spook then creates a new symling called `spook_inverse` from the custom_components root directory into the integrations folder.

tl;dr: remoce the symlink called `spook_inverse` in your `custom_components` folder and reload the integration.
